File: python/linked-lists/1721-swapping-nodes-in-a-linked-list.py
from typing import ListNode


def swapNodes(head: ListNode, k: int) -> ListNode:
    list_len = 0
    cur = head
    
    while cur:
        cur = cur.next
        list_len += 1
        
    left, right = None, None
    left_idx, right_idx = k - 1, list_len - k
    
    cur = head
    cur_idx = 0

    while cur:
        if cur_idx == left_idx:
            left = cur
        if cur_idx == right_idx:
            right = cur
            
        cur = cur.next
        cur_idx += 1
        
    left.val, right.val = right.val, left.val
    
    return head

File: js/linked-lists/1721-swapping-nodes-in-a-linked-list.js
var swapNodes = function(head, k) {
    var len = 0;
    var cur = head;
    while (cur) {
        cur = cur.next;
        len++;
    }
    
    var left = null;
    var right = null;
    var idx = 0;
    cur = head;
    for (var idx = 0; idx < len; idx++) {
        if (idx == k - 1) left = cur;
        if (idx == len - k) right = cur;
        cur = cur.next;
    }
    
    [left.val, right.val] = [right.val, left.val];
    
    return head;
};